docs: document WebAuthn conditional create (#44643)
mdn/content: files/en-us/web/api/web_authentication_api/index.md

> [!NOTE]
> Note that only [discoverable credentials](#discoverable_and_non-discoverable_credentials) are included in calls that use conditional mediation, because the browser needs to request applicable credentials without knowing the credential ID values for them.

### Automatic passkey creation

> Automatic passkey creation, also known as _conditional create_, lets an RP ask the browser to create a passkey after the user has already signed in with a password.

Reviewer: This is a curious statement: you don't need conditional create to "ask the browser to create a passkey". The point of course is to create a passkeys without asking the user first.

```suggestion
Automatic passkey creation, also known as _conditional create_, lets an RP ask the browser to create a passkey automatically, without asking the user, when certain conditions are met: specifically, when the user has previously agreed to create a credential and has recently signed in.
```

...or something like that.

Reviewer: But tbh I'm not exactly sure what the conditions are here. In this: https://developer.mozilla.org/en-US/docs/Web/Security/Authentication/Passkeys#conditional_create we say:

> If the user has just signed in with a password, using a password manager that also supports passkeys (that is, a credentials manager that can also function as an authenticator), then the browser will ask that credentials manager to create a new passkey for the user, without asking the user.

...that is, the user must have signed in using a password **stored in that credentials manager**, and this kind of makes sense, as we also say:

> The theory here is that if the user is relying on a credentials manager for sign-in already, then they implicitly trust it to look after their sign-in credentials in general, so they can trust it to create a new form of credential for them.

So that implies that if the user signs in with a password that isn't stored in a password manager, then the condition is not met.

The [spec](https://w3c.github.io/webappsec-credential-management/#dom-credentialmediationrequirement-conditional) says:

> a user has previously consented to credential creation and the user agent knows it recently mediated an authentication

...which is a bit hard to understand, but "consented to credential creation" presumably implies that the password was stored in a password manager, since that's what `create()` means with a password credential.

This can reduce passkey adoption friction: the user signs in as usual, and the browser can create a passkey when the user agent and passkey provider determine that the request is appropriate.

Use {{domxref("PublicKeyCredential.getClientCapabilities_static", "PublicKeyCredential.getClientCapabilities()")}} to check for the `conditionalCreate` capability before offering this flow.
If it is supported, call {{domxref("CredentialsContainer.create()")}} with:

- The usual [`publicKey`](/en-US/docs/Web/API/CredentialsContainer/create#publickey) creation options from the RP server.

> - [`mediation: "conditional"`](/en-US/docs/Web/API/CredentialsContainer/create#mediation).

Reviewer: ```suggestion
The `mediation` option set to [`"conditional"`](/en-US/docs/Web/API/CredentialsContainer/create#mediation).
```

```js
if (window.PublicKeyCredential?.getClientCapabilities) {
  const capabilities = await PublicKeyCredential.getClientCapabilities();

  if (capabilities.conditionalCreate) {
    const credential = await navigator.credentials.create({
      publicKey: optionsFromServer,
      mediation: "conditional",
    });

    // Send the credential to the server to complete registration.
  }
}
```

Call this flow soon after a successful password-based sign-in, while the user is still signed in.

> Conditional creation may fail if the required conditions are not met, so handle failures without showing errors to the user.

Reviewer: ```suggestion
Conditional creation will fail if the required conditions are not met, so handle failures without showing errors to the user.
```
